### PR TITLE
fix: use binary_to_existing_atom to replace some risky binary_to_atom

### PR DESCRIPTION
**Pull request by lafirest**
1.  emqx_bridge_resource
After investigation, it was confirmed that there was no need to convert the bridge name to atom.
We can keep it binary, whether the API, configuration or emqx_bridge all of these are supports names in binary format

2. emqx_bridge_api
When we use one API to send a request to a Node, it means that the Node is known, so it is better to use the `binary_to_existing_atom`
 And due to the maximum limit of the nested of the elvis, I had to adjust the function indent

3. emqx_connector
After investigation, it was confirmed that there was no need to convert the bridge name to atom.
We can keep it binary, whether the API, configuration both of these are supports names in binary format

4. Ensure that the node name is known

- emqx_dashboard_monitor_api
- emqx_gateway_api_clients
- emqx_mgmt_api_clients
- emqx_mgmt_api_subscriptions
When we use one API to send a request to a Node, it means that the Node is known, so it is better to use the `binary_to_existing_atom`

5. emqx_coap_pubsub_handler
The flag  for retain can only be true/false, which are two existing atoms

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

~~- [ ] Added tests for the changes~~
- [x] Changed lines covered in coverage report
- [x] Change log has been added to `changes/` dir
~~- [ ] For EMQX 4.x: `appup` files updated (execute `scripts/update-appup.sh emqx`)~~
- [x] For internal contributor: there is a jira ticket to track this change, and another jira tickt to track doc updates (if any)
       EMQX-8026
- [x] In case of non-backward compatible changes, reviewer should check this item as a write-off, and add details in **Backward Compatibility** section

## More information
This is one of a series of fixes for the atom attack risk, and there may be other PRs to follow, a PR per modification maybe could helps to describe in more detail why the change was made

### CODE REVIEW

File: apps/emqx_gateway/src/coap/handler/emqx_coap_pubsub_handler.erl [hunk 1]
```diff
@@ -121,7 +121,13 @@ apply_publish_opts(Msg, MQTTMsg) ->
             maps:fold(
                 fun
                     (<<"retain">>, V, Acc) ->
-                        Val = erlang:binary_to_atom(V),
+                        Val =
+                            case emqx_misc:safe_to_existing_atom(V) of
```
Comment (qzhuyan) [inline apps/emqx_gateway/src/coap/handler/emqx_coap_pubsub_handler.erl:125]: just match <<"true">> ?

File: apps/emqx/src/emqx_misc.erl [hunk 2]
```diff
@@ -533,6 +547,14 @@ readable_error_msg(Error) ->
             end
     end.
 
+try_to_existing_atom(Fun, Args) ->
+    try erlang:apply(Fun, Args) of
```
Comment (qzhuyan) [inline apps/emqx/src/emqx_misc.erl:551]: avoid use `erlang:apply/2`, you can try .. catch in L469

File: apps/emqx_gateway/src/coap/handler/emqx_coap_pubsub_handler.erl (hunk 1 above)
Comment (lafirest) [reply apps/emqx_gateway/src/coap/handler/emqx_coap_pubsub_handler.erl:125]: fix in #9285

File: apps/emqx/src/emqx_misc.erl (hunk 2 above)
Comment (lafirest) [reply apps/emqx/src/emqx_misc.erl:551]: fix in #9285

**Pull request merged**